src/engine/IndexScan.h

```diff
@@ -292,6 +292,12 @@ class IndexScan final : public Operation {
   std::optional<std::shared_ptr<QueryExecutionTree>>
   makeTreeWithStrippedColumns(
       const std::set<Variable>& variables) const override;
+
+  // Return a shared pointer to the correct permutation in `index` based on the
+  // `permutation` enum and the values in `triple`.
```

Comment:
be more specific (if any of the subject, predicate, or object are "internal", then return the internal permutation.